compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertConv2D1x1ToMatmulPass.cpp

```diff
@@ -27,17 +27,28 @@ class Convert1x1ConvolutionMatmulOp
 
   LogicalResult matchAndRewrite(linalg::Conv2DNhwcHwcfOp convOp,
                                 PatternRewriter &rewriter) const override {
-    ShapedType inputShapeType =
-        convOp.getInputOperand(0)->get().getType().cast<ShapedType>();
-    ShapedType filterShapeType =
-        convOp.getInputOperand(1)->get().getType().cast<ShapedType>();
-    ShapedType outputShapeType =
-        convOp.getOutputOperand(0)->get().getType().cast<ShapedType>();
+    RankedTensorType inputShapeType =
+        convOp.getInputOperand(0)->get().getType().dyn_cast<RankedTensorType>();
+    RankedTensorType filterShapeType =
+        convOp.getInputOperand(1)->get().getType().dyn_cast<RankedTensorType>();
+    RankedTensorType outputShapeType = convOp.getOutputOperand(0)
+                                           ->get()
+                                           .getType()
+                                           .dyn_cast<RankedTensorType>();
+
+    if (!inputShapeType || !filterShapeType || !outputShapeType)
+      return failure();
 
     auto inputShape = inputShapeType.getShape();
     auto filterShape = filterShapeType.getShape();
     auto outputShape = outputShapeType.getShape();
 
+    bool inputDynWidthHeight = inputShape[1] == ShapedType::kDynamicSize &&
+                               inputShape[2] == ShapedType::kDynamicSize;
+
+    // We cannot collapse if the width and heigh are dynamic.
```

Comment:
Not fully following everything here, but we can collapse when width and height are dynamic....

Reply:
Updated the comment. The problem is not the collapse but rather `Expand` does not support `? -> ?x?` to preserve both dynamic dimensions.